### PR TITLE
Feature/Improvements on import and render

### PR DESCRIPTION
**Pull request by lucaswilliameufrasio**
## Description

- Removes the cyclic dependency between components and interfaces;
- Split tsconfig into two configurations, one for build and one for development;
- Move tests to their own folder.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [X] Created a React application and used this library in it.
- [X] Ran all tests of this application.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/CanRender.spec.tsx
Comment (icastro085) [inline src/CanRender.spec.tsx]: What is the difference between old approach and this new?

File: src/CanRender.tsx
Comment (icastro085) [inline src/CanRender.tsx]: This approach is necessary here? What do you think about to use the old import format?

File: src/ElseCanRender.tsx
Comment (icastro085) [inline src/ElseCanRender.tsx]: Here is the same question, why don't use the old approach?

File: src/ElseRender.tsx
Comment (icastro085) [inline src/ElseRender.tsx]: Here is the same question, why don't use the old approach?

File: src/ElseRenderContent.tsx
Comment (icastro085) [inline src/ElseRenderContent.tsx]: My question here is if this is necessary to have another file to render the content

File: tsconfig-build.json
Comment (icastro085) [inline tsconfig-build.json]: need to have a empty line

**Review by icastro085 — CHANGES_REQUESTED** [review]
I have some doubts about the tests and why did you change the approach? My main question is the difference about it!!
Other question is about the layze load, for me the file is so small que the layze load is not necessary, what do you think?

File: src/CanRender.spec.tsx
Comment (lucaswilliameufrasio) [reply src/CanRender.spec.tsx]: You are talking about the waitFor or the destructuring of queryByText from the render method?

Comment (lucaswilliameufrasio) [reply src/CanRender.spec.tsx]: If it is about the destructuring, there is no difference.

File: src/CanRender.tsx
Comment (lucaswilliameufrasio) [reply src/CanRender.tsx]: My idea was to only import this component when the component was rendering or in this case, when the condition passed to allow rendering were true but looking at a big picture, this is doing nothing there because the component that needs to be lazy-loaded is the component passed as children to CanRender, ElseCanRender, and ElseRender. So I will revert to the old approach.

Comment (lucaswilliameufrasio) [reply src/CanRender.tsx]: Solved.

File: src/ElseCanRender.tsx
Comment (lucaswilliameufrasio) [reply src/ElseCanRender.tsx]: Solved.

File: src/ElseRenderContent.tsx
Comment (lucaswilliameufrasio) [reply src/ElseRenderContent.tsx]: It's not, I've just removed it.

File: tsconfig-build.json
Comment (lucaswilliameufrasio) [reply tsconfig-build.json]: Done.

**Comment by lucaswilliameufrasio** [comment]
At least I've got some improvements on unpacked size. NPM reports this library has 76kB when unpacked, now it has 12.2kB.

image

**Comment by lucaswilliameufrasio** [comment]
The cyclic dependency still exists...

**Comment by lucaswilliameufrasio** [comment]
I managed to remove the cyclic dependency using `defaultProps`.

Unfortunately, this increased the unpackaged size and now it is 13.5kB.

image

File: src/helpers/index.ts
Comment (icastro085) [inline src/helpers/index.ts]: Add here the returned type

File: src/helpers/index.ts
Comment (icastro085) [inline src/helpers/index.ts]: Add a blank line to separete the "if"

**Review by icastro085 — APPROVED** [review]
There are some comments, but it doesn't block the merge. If you change before that I merge it, I will thankful

File: src/helpers/index.ts
Comment (lucaswilliameufrasio) [reply src/helpers/index.ts]: Done.

File: src/helpers/index.ts
Comment (lucaswilliameufrasio) [reply src/helpers/index.ts]: Done.

**Comment by lucaswilliameufrasio** [comment]
Already did the things you requested me to do on sunday.

**Review by icastro085 — APPROVED** [review]

**Pull request merged**
